Module 3 block 2 : Math Magicians: tests

aceDavon opened the pull request:
## We have Implemented the following tasks in this PR: :writing_hand: 

- [ ] Set up React Testing Library
- [ ] Follow the instructions in the official documentation
- [ ] Write unit tests for the files operate.js and calculate.js using Jest
- [ ] Create unit tests for all React components:
 - Use Jest snapshots to test the components
 - Use React Testing Library to simulate user interaction
- [ ] Run full tests suite using Jest

aceDavon committed:
Create Test suites for Logic
 Create and implement test suites for both the Operate.js and the
 Calculate.js functions

tarikbouari committed:
Create test for components

tarikbouari committed:
Create tests for UI component

tarikbouari committed:
update file

tarikbouari committed:
fix linters errors

GzimAsani commented on src/logic/logic.test.js:
```diff
@@ -0,0 +1,18 @@
+import calculate from './Calculate';
+import operate from './Operate';
+
+describe('testing logic components', () => {
+  test('Test the calculator logic', () => {
+    const data = {
+      total: 4,
+      next: 3,
+      operation: null,
+    };
+    const value = { total: 3, next: null, operation: '+' };
+    expect(calculate(data, '+')).toEqual(value);
+  });
+  test('Check operate for dividion by 0', () => {
+    const str = "Can't divide by 0.";
+    expect(operate(4, 0, '÷')).toBe(str);
+  });
+});
```
- [ ] You should cover all the scenarios on operate and calculate components. Besides adding and division you should also test subtraction and multiplying

GzimAsani replied:
- [ ] Please use `React Testing Library` to simulate user interaction.

GzimAsani reviewed (changes requested):
### Hello @aceDavon and @tarikbouari             :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, you did most of the work +1: 
 
 ## Required Changes 

- [ ]  Update readme file and explain to the potential users how can they test your project
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

aceDavon committed:
Add test implementations

Add tests implementation to the logic test suite to test for other
operate functionalites other than the addition and division
funtionalities

aceDavon committed:
Add test implementation documentation

Add a description of the test implementation and explain how to run the
tests suites already written for the project

aceDavon committed:
Add contributions to documentation

Add contributing partners to the README.md documentation

aceDavon committed:
Implement user Interaction tests

Implement unit testing to mock user interactions

TedLivist reviewed (approved):
## STATUS: APPROVED ✔️ ✔️

Hi @aceDavon and @tarikbouari,

✔️ All tests are passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
